.github/workflows/go.yml

```diff
@@ -157,6 +157,43 @@ jobs:
 #          codeCoverage=${{ env.CODE_COVERAGE }}
 #          codeCoverage=${codeCoverage%??}
 #          if [[ $codeCoverage -lt 92 ]]; then echo "code coverage cannot be less than 92%, currently its ${{ env.CODE_COVERAGE }}%" && exit 1; fi;
+  Submodule-Unit-Testing:
+    name: Submodule Unit Testing (v${{ matrix.go-version }})🛠
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go-version: [ '1.23', '1.22' ]
+
+    steps:
+      - name: Checkout code into go module directory
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
+        id: Go
+
+      - name: Detect and Test Submodules
+        run: |
+          # Find all directories containing a go.mod file within 'pkg'
+          for module in $(find pkg -name "go.mod" -exec dirname {} \;); do
+            echo "Testing module: $module"
+            cd $module
+
+            # Download dependencies for the submodule
+            go mod download
+            go mod tidy
+
+            # Run tests for the submodule and generate coverage
+            export APP_ENV=test
+            go test ./... -v -short -coverprofile=submodule.cov -coverpkg=./...
```

Comment:
I'm surprised by the fact you generate coverage files named `submodule.cov`, but I don't see code to merge them with the others

Reply:
addressed the issue, the coverage reports are uploaded to the artifacts in the updates code.

Thanks for pointing out !